components/esp_modem/examples/modem_console/main/modem_console_main.cpp

```diff
@@ -145,6 +179,26 @@ extern "C" void app_main(void)
     esp_console_dev_uart_config_t uart_config = ESP_CONSOLE_DEV_UART_CONFIG_DEFAULT();
     ESP_ERROR_CHECK(esp_console_new_repl_uart(&uart_config, &repl_config, &s_repl));
 
+    switch (esp_sleep_get_wakeup_cause()) {
+        case ESP_SLEEP_WAKEUP_TIMER:
+            if(esp_modem::modem_mode::CMUX_MODE == mode_rtc){
+                ESP_LOGI(TAG, "Deep sleep reset\n");
+
+                /* Set APN */
+                auto new_pdp = std::unique_ptr<PdpContext>(new PdpContext(apn_rtc));
+                dce->get_module()->configure_pdp_context(std::move(new_pdp));
+
+                /* Set CMUX mode */
+                if (!dce->set_mode(mode_rtc)) {
+                    ESP_LOGE(TAG, "Failed to set the desired mode");
+                }
+            }
+            break;
+        case ESP_SLEEP_WAKEUP_UNDEFINED:
+        default:
+            ESP_LOGI(TAG, "Not a deep sleep reset\n");
```

Comment:
Since I assume this example will be mostly used without deep sleep
```suggestion
            ESP_LOGD(TAG, "Not a deep sleep reset\n");
```